pages/projects/[id]/edit.js

```diff
@@ -0,0 +1,58 @@
+import { Router, useRouter } from "next/router";
+import Form from "@/components/Form";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+
+export default function Edit({ projects, setNewProjects }) {
+  const [formMaterials, setFormMaterials] = useState([""]);
+
+  const [formSteps, setFormSteps] = useState([{ id: "1", description: "" }]);
+
+  const router = useRouter();
+  const { id } = router.query;
+
+  const projectData = projects.find((project) => project.id === id);
+
+  useEffect(() => {
+    if (projectData) {
+      setFormMaterials(projectData.materials);
+      setFormSteps(projectData.steps);
+    }
+  }, [projects]);
+
+  function editProject(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const newProject = Object.fromEntries(formData);
+    newProject.id = id;
+    newProject.materials = formMaterials;
+    newProject.steps = formSteps;
+    const updatedProject = projects.map((project) =>
+      project.id === id ? newProject : project
+    );
+    setNewProjects(updatedProject);
```

Comment:
The updating of all existing projects should not be part of the same function that extracts the data for the updated project.

Instead, you should have a helper function in your `_app.js` that takes care of that and is being passed down as a prop to this page and called within the editProject function.

```suggestion
   const updatedProject = Object.fromEntries(formData);
    updatedProject.id = id;
    updatedProject.materials = formMaterials;
    updatedProject.steps = formSteps;
    onUpdateProject(updatedProject);
```

The corresponding function in your `_app.js` could look like this:

```js
function handleUpdateProject(updatedProject) {
    setNewProjects(
      newProjects.map((project) => {
        return project.id === updatedProject.id ? updatedProject : project;
      })
    );
  }
```